NXP MCX SOCs have hard dependency on SEGGER debug module

**Describe the bug**
NXP MCX Series SOCs have a hard dependency on the SEGGER debug module- if a downstream west workspace is created without the SEGGER debug module included, the Kconfig phase of the build will fail.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?` frdm_mcxn947//cpu0`

**To Reproduce**
This can be reproduced on the tip of main, 9f93dede369c365c. Edit the `west.yml` file with the following change:

```diff
diff --git a/west.yml b/west.yml
index 5d50af43600..bd19ed8ce82 100644
--- a/west.yml
+++ b/west.yml
@@ -316,11 +316,6 @@ manifest:
     - name: picolibc
       path: modules/lib/picolibc
       revision: d492d5fa7c96918e37653f303028346bb0dd51a2
-    - name: segger
-      revision: b011c45b585e097d95d9cf93edf4f2e01588d3cd
-      path: modules/debug/segger
-      groups:
-        - debug
     - name: tinycrypt
       revision: 1012a3ebee18c15ede5efc8332ee2fc37817670f
       path: modules/crypto/tinycrypt
```
Then, build a sample app like `hello_world` for the `frdm_mcxn947//cpu0` target:

`west build -p always -b frdm_mcxn947//cpu0 samples/hello_world`

**Expected behavior**
It should be possible to build this SOC without the SEGGER module

**Impact**
Downstream Zephyr developers must include this module in their workspace in order to develop with MCX series SOCs

**Logs and console output**
```
error: Aborting due to Kconfig warnings

CMake Error at /home/danieldegrasse/zephyr/zephyrproject/zephyr/cmake/modules/kconfig.cmake:395 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /home/danieldegrasse/zephyr/zephyrproject/zephyr/cmake/modules/zephyr_default.cmake:133 (include)
  /home/danieldegrasse/zephyr/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/danieldegrasse/zephyr/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /usr/bin/cmake -DWEST_PYTHON=/home/danieldegrasse/zephyr/.venv/bin/python3 -B/home/danieldegrasse/zephyr/zephyrproject/zephyr/build -GNinja -DBOARD=frdm_mcxn947//cpu0 -S/home/danieldegrasse/zephyr/zephyrproject/zephyr/samples/hello_world
```
**Environment (please complete the following information):**

 - OS: Ubuntu 22.04 LTS
 - Toolchain Zephyr SDK
 - Commit SHA or Version used: 9f93dede369c365c with patch applied as described above

--- a/soc/nxp/mcx/mcxa/Kconfig
+++ b/soc/nxp/mcx/mcxa/Kconfig
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config SOC_SERIES_MCXA
-	select HAS_SEGGER_RTT
+	select HAS_SEGGER_RTT if ZEPHYR_SEGGER_MODULE
 	select CLOCK_CONTROL
 	select ARM
 	select HAS_MCUX

--- a/soc/nxp/mcx/mcxn/Kconfig
+++ b/soc/nxp/mcx/mcxn/Kconfig
@@ -2,7 +2,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config SOC_SERIES_MCXN
-	select HAS_SEGGER_RTT
+	select HAS_SEGGER_RTT if ZEPHYR_SEGGER_MODULE
 	select CLOCK_CONTROL
 	select ARM
 	select HAS_MCUX

